WordPressKitTests/Utilities/HTTPRequestBuilderTests.swift

```diff
@@ -0,0 +1,93 @@
+import Foundation
+import XCTest
+
+@testable import WordPressKit
+
+class HTTPRequestBuilderTests: XCTestCase {
+
+    func testDefaultMethod() throws {
+        let request = try HTTPRequestBuilder(url: URL(string: "https://wordpress.org")!).build()
+        XCTAssertEqual(request.url?.absoluteString, "https://wordpress.org")
+        XCTAssertEqual(request.httpMethod, "GET")
+    }
```

Comment:
This test made me want to suggest to triangulate setting the URL, that is add at least two different examples (cc @wordpress-mobile/mobile-ui-testing-squad)

```swift
XCTAssertEqual(
  try HTTPRequestBuilder(url: URL(string: "https://wordpress.org/")!).build().url?.absoluteString,
  "https://wordpress.org/"
)
XCTAssertEqual(
  try HTTPRequestBuilder(url: URL(string: "https://wordpress.com/")!).build().url?.absoluteString,
  "https://wordpress.com/"
)
```

The value of this approach is that it gives additional confidence in the robustness of the tests. **But that's mostly redundant when we have more tests below that implicitly exercise this behavior.**

However, something I noticed and that we could use to add a variety of initial `URL` inputs is that there are no tests for how `set(method:)` behaves. Granted, that's pretty straightforward, but given how core this object will be, I think "overtesting" it wouldn't be a waste. The same goes for the `HTTPRequestBuilder.Method` `allowsHTTPBody` logic.